proxy/envoy/config.go

```diff
@@ -607,13 +612,40 @@ func buildClusters(services []*model.Service) []Cluster {
 				Ports:    []*model.Port{port},
 				Tags:     svc.Tags,
 			}
+
 			cluster := Cluster{
 				Name:             OutboundClusterPrefix + clusterSvc.String(),
 				ServiceName:      clusterSvc.String(),
 				Type:             "sds",
 				LbType:           DefaultLbType,
 				ConnectTimeoutMs: DefaultTimeoutMs,
 			}
+
+			if up, _ := upstreamsByHostname[svc.Hostname]; up != nil {
+				cluster.LbType = loadBalanceType(up.GetLbPolicy())
+				if cb := up.GetCircuitBreaker(); cb != nil {
+					if scb := cb.GetSimpleCb(); scb != nil {
+						cluster.OutlierDetection = &OutlierDetection{
+							ConsecutiveError:   int(scb.GetFailureThreshold()),
+							BaseEjectionTimeMS: int(1000 * scb.GetResetTimeoutSeconds()),
+						}
+					}
+				}
+
+				if t := up.GetTimeout(); t != nil {
+					if st := t.GetSimpleTimeout(); st != nil {
+						cluster.ConnectTimeoutMs = int(st.GetTimeoutSeconds() * 1000)
+					}
+				}
+
+				if r := up.GetRetry(); r != nil {
+					if sr := r.GetSimpleRetry(); sr != nil {
+						cluster.CircuitBreaker = &CircuitBreaker{
+							MaxRetries: int(sr.GetAttempts()),
+						}
+					}
+				}
+			}
```

Comment:
these two should go into route blocks associated with the cluster. @GregHanson did something similar in a8. We might need an inverse map from clusters to associated routes (just like we have routes to clusters in resources.go).